confirm/confirm.go

```diff
@@ -0,0 +1,206 @@
+// Package confirm provides an interface to ask a user to confirm an action.
+// The user is provided with an interface to choose an affirmative or negative
+// answer, which is then reflected in the exit code for use in scripting.
+//
+// If the user selects the affirmative answer, the program exits with 0. If the
+// user selects the negative answer, the program exits with 1.
+//
+// I.e. confirm if the user wants to delete a file
+//
+// $ gum confirm "Are you sure?" && rm file.txt
+package confirm
+
+import (
+	"time"
+
+	"github.com/charmbracelet/bubbles/help"
+	"github.com/charmbracelet/bubbles/key"
+	"github.com/charmbracelet/gum/timeout"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+)
+
+func defaultKeymap(affirmative, negative string) keymap {
+	return keymap{
+		Abort: key.NewBinding(
+			key.WithKeys("ctrl+c"),
+			key.WithHelp("ctrl+c", "cancel"),
+		),
+		Quit: key.NewBinding(
+			key.WithKeys("esc"),
+			key.WithHelp("esc", "quit"),
+		),
+		Negative: key.NewBinding(
+			key.WithKeys("n", "N", "q"),
+			key.WithHelp("n", negative),
+		),
+		Affirmative: key.NewBinding(
+			key.WithKeys("y", "Y"),
+			key.WithHelp("y", affirmative),
+		),
+		Toggle: key.NewBinding(
+			key.WithKeys(
+				"left",
+				"h",
+				"ctrl+n",
+				"shift+tab",
+				"right",
+				"l",
+				"ctrl+p",
+				"tab",
+			),
+			key.WithHelp("←/→", "toggle"),
+		),
+		Submit: key.NewBinding(
+			key.WithKeys("enter"),
+			key.WithHelp("enter", "submit"),
+		),
+		Help: key.NewBinding(
+			key.WithKeys("?"),
+			key.WithHelp("?", "toggle help"),
+		),
+	}
+}
+
+type keymap struct {
+	Abort       key.Binding
+	Quit        key.Binding
+	Negative    key.Binding
+	Affirmative key.Binding
+	Toggle      key.Binding
+	Submit      key.Binding
+	Help        key.Binding
+}
+
+// FullHelp implements help.KeyMap.
+func (k keymap) FullHelp() [][]key.Binding {
+	return [][]key.Binding{
+		{k.Toggle, k.Submit, k.Affirmative, k.Negative},
+		{k.Abort, k.Quit, k.Help},
+	}
+}
```

Comment:
i wonder if need full help? didn't add to the other components, I think we either add it in all or none of them, ideally?

wdyt @meowgorithm

Reply:
Hmm. I think you're right that it's generally extraneous, though in some cases I think it's helpful. Why don't we hide it by default across the board and eventually aim to get it into all the components that make sense?